api-tests/src/steps/ipv-steps.ts

```diff
@@ -315,11 +317,24 @@ Then(
       throw new Error("No identity found.");
     }
 
-    const namePart = this.identity[identityCredential].name?.[0].nameParts.find(
-      (np) => {
-        return field === np.type;
-      },
-    );
+    if (!this.vcs) {
+      throw new Error("No credentials found.");
+    }
+
+    const inheritedIdentity = this.vcs[
+      "https://orch.stubs.account.gov.uk/migration/v1"
+    ]
+      ? (this.vcs["https://orch.stubs.account.gov.uk/migration/v1"]
+          .vc as IdentityCheckCredentialClass)
+      : undefined;
+
+    const namePart = (
+      inheritedIdentity
+        ? (inheritedIdentity.credentialSubject as IdentityCheckSubjectClass)
+        : this.identity[identityCredential]
+    ).name?.[0].nameParts.find((np) => {
+      return field === np.type;
+    });
```

Comment:
This change shouldn't really be necessary - suggests there might be an issue in the way we find the name :(

Which scenario break if you don't do this?

Reply:
If it is broken, we might need to raise a separate ticket (unless it's a very easy fix)

Reply:
note from slack: this was bc in UserIdentityService, the inherited identity with no evidence was being marked as unsuccessful bc it had an empty evidence block so we mark inherited identities as successful